afids_utils/tests/strategies.py

```diff
@@ -0,0 +1,95 @@
+from __future__ import annotations
+
+import numpy as np
+from hypothesis import strategies as st
+from hypothesis.extra.numpy import arrays
+from numpy.typing import NDArray
+
+
+@st.composite
+def afid_coords(
+    draw: st.DrawFn,
+    min_value: float = -50.0,
+    max_value: float = 50.0,
+    width: int = 16,
+) -> NDArray[np.single]:
+    return draw(
+        arrays(
+            shape=(32, 3),
+            dtype=np.single,
+            elements=st.floats(
+                min_value=min_value, max_value=max_value, width=width
+            ),
+        )
+    )
+
+
+@st.composite
+def world_coord(
+    draw: st.DrawFn,
+    min_value: float = -50.0,
+    max_value: float = 50.0,
+    width: int = 16,
+) -> NDArray[np.single]:
+    return draw(
+        arrays(
+            shape=(3, 1),
+            dtype=np.single,
+            elements=st.floats(
+                min_value=min_value, max_value=max_value, width=width
+            ),
+        )
+    )
+
+
+@st.composite
+def affine_xfm(
+    draw: st.DrawFn,
+) -> NDArray[np.single]:
+    affine = np.eye(4)
+
+    rotation_vals = draw(
+        arrays(
+            shape=(3),
+            dtype=np.float_,
+            elements=st.floats(min_value=-np.pi, max_value=np.pi),
+        )
+    )
+    rotation_x = np.array(
+        [
+            [1, 0, 0, 0],
+            [0, np.cos(rotation_vals[0]), -np.sin(rotation_vals[0]), 0],
+            [0, np.sin(rotation_vals[0]), np.cos(rotation_vals[0]), 0],
+            [0, 0, 0, 1],
+        ]
+    )
+    rotation_y = np.array(
+        [
+            [np.cos(rotation_vals[1]), 0, np.sin(rotation_vals[1]), 0],
+            [0, 1, 0, 0],
+            [-np.sin(rotation_vals[1]), 0, np.cos(rotation_vals[1]), 0],
+            [0, 0, 0, 1],
+        ]
+    )
+    rotation_z = np.array(
+        [
+            [np.cos(rotation_vals[2]), -np.sin(rotation_vals[2]), 0, 0],
+            [np.sin(rotation_vals[2]), np.cos(rotation_vals[2]), 0, 0],
+            [0, 0, 1, 0],
+            [0, 0, 0, 1],
+        ]
+    )
+    rotation = rotation_x @ rotation_y @ rotation_z
+
+    translation = draw(
+        arrays(
+            shape=(3),
+            dtype=np.float_,
+            elements=st.floats(min_value=-50.0, max_value=50.0),
+        )
+    )
+
+    affine[:3, :3] = rotation[:3, :3]
+    affine[:3, 3] = rotation[:3, :3] @ translation + rotation[:3, 3]
```

Comment:
I haven't really looked it up so it probably works but what is going on here? I think my preference would just be to make a 4x4 translation matrix and multiply it by the rotation matrix. We could even randomize the order in which the matrices are multiplied.

Reply:
Ahh, actually looking at this, I think this is over complicated. It was applying the rotation to the translation as well, but we really only need to generate a single 4x4 matrix (which would also eliminate the need to do any intermediate matrix multiplication / randomizing the order of matrix multiplication.